SoftLayer/managers/hardware.py

```diff
@@ -319,11 +319,11 @@ def change_port_speed(self, hardware_id, public, speed):
         if public:
             return self.client.call('Hardware_Server',
                                     'setPublicNetworkInterfaceSpeed',
-                                    speed, id=hardware_id)
+                                    [rebundant, speed], id=hardware_id)
```

Comment:
redundant should be the second parameter, not the first. https://sldn.softlayer.com/reference/services/SoftLayer_Hardware_Server/setPublicNetworkInterfaceSpeed/